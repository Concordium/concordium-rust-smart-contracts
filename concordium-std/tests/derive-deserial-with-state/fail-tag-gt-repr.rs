//! Ensure `derive(DeserialWithState)` fails when a 'tag' attribute is greater
//! than what 'repr(u8)' can represent.
use concordium_std::*;

#[derive(DeserialWithState)]
#[concordium(repr(u8))]
#[concordium(state_parameter = "S")]
enum MyTaggedEnum<S> {
    #[concordium(tag = 256)]
    One(StateSet<u32, S>),
    Two,
}

fn main() {}
