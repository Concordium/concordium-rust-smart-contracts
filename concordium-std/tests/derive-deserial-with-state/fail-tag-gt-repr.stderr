error: 'tag' attribute value of 256 cannot be represented by the type u8 set in 'repr(u8)'
 --> tests/derive-deserial-with-state/fail-tag-gt-repr.rs:9:18
  |
9 |     #[concordium(tag = 256)]
  |                  ^^^
